superset/connectors/base/models.py

```diff
@@ -339,11 +339,14 @@ def data_for_slices(  # pylint: disable=too-many-locals
                     or []
                 )
             else:
-                column_names.update(
-                    column
+                _columns = [
```

Comment:
Is this extra variable really needed?

Reply:
Since code style limits the column width to 88 characters, not adding an extra variable will make the commits line larger. So I added an extra variable to make it more readable.

```
diff --git a/superset/connectors/base/models.py b/superset/connectors/base/models.py
index b888f4afc..cc42d4e7f 100644
--- a/superset/connectors/base/models.py
+++ b/superset/connectors/base/models.py
@@ -339,13 +339,17 @@ class BaseDatasource(
                     or []
                 )
             else:
-                column_names.update([
-                    utils.get_column_name(column)
-                    if utils.is_adhoc_column(column)
-                    else column
-                    for column_param in COLUMN_FORM_DATA_PARAMS
-                    for column in utils.get_iterable(form_data.get(column_param) or [])
-                ])
+                column_names.update(
+                    [
+                        utils.get_column_name(column)
+                        if utils.is_adhoc_column(column)
+                        else column
+                        for column_param in COLUMN_FORM_DATA_PARAMS
+                        for column in utils.get_iterable(
+                            form_data.get(column_param) or []
+                        )
+                    ]
+                )

         filtered_metrics = [
             metric
```